content/essentials/refs.mdx

```diff
@@ -0,0 +1,106 @@
+# Refs
+
+Refs, or references, allow you to have direct access to DOM element. 
+
+To create a `ref` in Solid, you can get a reference to an element through assigning it to avariable: 
+
+```jsx
+const myDiv = <div>My Element</div>;
+```
+
+For a more idiomatic way, Solid provides the `ref` attribute. 
+Refs in Solid are assignments that occur at creation time before elements are attached to the document DOM.
+To use this, you simply need to declare a variable and use it as a ref attribute:
+
+```tsx
+let myDiv;
+
+<div ref={myDiv}>My Element</div>
+
+```
+
+This lets you access its properties and methods directly, allowing for manipulatinos such as animations and event handling.
+
+## Callbacks
+
+Refs can also take the form of a callback function.
+This method can be convenient for encapsulating logic, especially when there is no need to wait until the elements are attached:
+
+```tsx
+let myDiv;
+
+<div ref={(el) => myDiv = el}>My Element</div>
+```
+
+Callback refs provide a more flexible way to set refs and can be beneficial in various scenarious, such as when you need to attach event listeners to the DOM elements immediately after creation.
+
+## Forwarding refs
+
+Forwarding refs provide a way to expose one of the inner elements' references to a parent component.
+By doing this, the parent component can directly access and interact the inner elements.
+
+When a component receives a `ref` attribute from its parent, it will come as `props.ref` in a callback form, regardless of whether the parent passed it as a simple assignment or a callback.
```

Comment:
To help explain this, maybe also show ref={(el) => props.ref(el)} (which can be shortened to the example below)

Reply:
could you explicitly show me (I'm real out of it today tbf lol)

Reply:
```ts
const MyComponent = (props: {
  ref: HTMLDivElement | ((el: HTMLDivElement) => void);
}) => {
  return (
    <div ref={(el) => (props.ref as (el: HTMLDivElement) => void)(el)}>
      My Element
    </div>
  );
};

let myRef!: HTMLDivElement;

<MyComponent ref={myRef} />;
```